lib/nghttp2_buf.h

```diff
@@ -25,9 +25,7 @@
 #ifndef NGHTTP2_BUF_H
 #define NGHTTP2_BUF_H
 
-#ifdef HAVE_CONFIG_H
-#  include <config.h>
-#endif /* HAVE_CONFIG_H */
+#include "nghttp2_config.h"
```

Comment:
I like better the pattern mbedtls has: 
```
#ifdef NGHTTP2_USER_CONFIG
#    include "nghttp2/config.h"
#endif
```
Keeping the compiler directives means we don't have to have a config file or run cmake. What do you think?

Reply:
Then i will have same issue as with mbedtls :D

Reply:
https://github.com/ARMmbed/mbedtls/pull/2459